### ISSUE TITLE
Can't get reopened indices with wildcards

### ISSUE THREAD

**Issue by mpfz0r** [Graylog2/graylog2-server#10572]
The indices resource tries to find all reopened indices by using
a wildcard. E.g. `restored-archive-*reopened`

With ES6 this doesn't work at all, because the code resolves aliases manually and the match
is ignoring wildcards:
https://github.com/Graylog2/graylog2-server/blob/master/graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/IndicesAdapterES6.java#L196

With ES7 this does work, because it uses the ES `_alias` api.
https://github.com/Graylog2/graylog2-server/blob/master/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndicesAdapterES7.java#L137
However, it triggers an exception as soon as there are multiple indices.
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java#L148

## Expected Behavior
Both ES versions should return all aliases for wildcards

## Current Behavior
ES6 doesn't
ES7 only works for one alias

## Possible Solution

Change the `resolveAlias()` API to reflect how ES aliases are structured.

## Steps to Reproduce (for bugs)

1. Restore multiple archives
2. Call `https://127.0.0.1:9000/api/system/indexer/indices/reopened` endpoint
3. Either receive an exception or no result

**Comment by mpfz0r** [comment]
One more bug:
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java#L258
This doesn't work with wildcards either

**Comment by mpfz0r** [comment]
Here's also a test that triggers the problem:
```--- graylog2-server/src/test/java/org/graylog2/indexer/indices/IndicesIT.java
+++ graylog2-server/src/test/java/org/graylog2/indexer/indices/IndicesIT.java
@@ -186,6 +186,17 @@ public abstract class IndicesIT extends ElasticsearchBaseTest {
         assertThat(indices.aliasTarget(alias)).contains(index);
     }
 
+    @Test
+    public void aliasTargetReturnsListOfTargetsGivenAliasIsPointingToWithWildcards() {
+        final String index = client().createRandomIndex("indices_it_");
+        final String alias = "graylog_alias_target";
+        assertThat(indices.aliasTarget(alias)).isEmpty();
+
+        client().addAliasMapping(index, alias);
+
+        assertThat(indices.aliasTarget("graylog_alias_*")).contains(index);
+    }
+
```

**Comment by mpfz0r** [comment]
@bernd what's the state of this? I think this should be fix before 4.1
